text/0026-rest-api/rest-api.yaml

```diff
@@ -0,0 +1,707 @@
+              schema:
+                $ref: '#/components/schemas/GetIsConfirmedResponse'
+              examples:
+                Example 1:
+                  $ref: '#/components/examples/get-is-confirmed-response-example-1'
+                Example 2:
+                  $ref: '#/components/examples/get-is-confirmed-response-example-2'
+        '400':
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+        '404':
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/can-not-find-data-response-example'
+  /messages:
+    post:
+      tags:
+        - messages
+      summary: Submit a message.
+      description: >-
+        Submit a message. The node takes care of missing fields and tries to
+        build the message. On success, the message will be stored in the Tangle.
+        This endpoint will return the identifier of the message by the response
+        headers.
+      operationId: post-messages
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/PostMessagesParameters'
+        required: true
+      responses:
+        '201':
+          description: successful operation
+          headers:
+            Location:
+              schema:
+                $ref: '#/components/schemas/PostMessagesResponse'
+              description: Identifier of the submitted message.
+        '400':
+          description: unsuccessful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+              examples:
+                default:
+                  $ref: '#/components/examples/invalid-data-provided-response-example'
+  '/outputs/by-id/{outputId}':
+    get:
+      tags:
+        - UTXO
+      summary: Find an output by its identifier.
+      description: >-
+        Find an output by its identifier. This endpoint returns ID of the
+        transaction that produced the output, the corresponding output index,
+        the address, the amount and the spend status. This information can only
+        be retrieved for outputs which are part of a confirmed transaction.
+      operationId: get-outputs-by-id
+      parameters:
+        - in: path
+          name: outputId
+          schema:
+            type: string
+          required: true
+          description: >-
+            Identifier of the output. An output is identified by the
+            concatenation of `transaction_id+output_index`.
```

Comment:
A sub-selector such as `output_index` would probably be most RESTful (and readeable) as another path parameter, so that: `/outputs/by-id/{transaction_id}/{output_index}`.

Reply:
The above also makes sense because you return these two separately as a result of other requests too, and therefore a concatenated selector like you propose is already a new _type_ on its own, harder to validate as well.